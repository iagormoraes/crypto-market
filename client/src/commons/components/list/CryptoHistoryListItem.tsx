import Image from 'next/image';

import { CryptoAsset } from '../../../modules/crypto/interfaces/asset.interface';

import { difference } from '../../utils/difference.util';

export default function CryptoHistoryListItem({
  cryptoHistory,
}: {
  cryptoHistory: CryptoAsset;
}) {
  return (
    <div className="p-2 bg-blueGray-500 hover:bg-blueGray-400 transition-colors grid grid-cols-4 items-center gap-2 text-white">
      <div>
        <div className="flex items-center gap-2">
          <div className="w-8 h-8 cover-img rounded-full overflow-hidden shrink-0">
            <Image src="/bitcoin.jpg" alt="bitcoin" layout="fill" />
          </div>
          <h4 className="font-bold text-lg text-white">
            {cryptoHistory.assetCode}
          </h4>
        </div>
        <p>
          <strong>Low</strong>: {cryptoHistory.low}
        </p>
        <p>
          <strong>High</strong>: {cryptoHistory.high}
        </p>
      </div>
      <div>
        <p>
          <strong>Ask</strong>: ${cryptoHistory.ask}
        </p>
        <p>
          <strong>Bid</strong>: ${cryptoHistory.bid}
        </p>
      </div>
      <div>
        <p>
          <strong>Close</strong>: {cryptoHistory.close}
        </p>
        <p>
          <strong>Open</strong>: {cryptoHistory.open}
        </p>
      </div>
      <div>
        <p>
          <strong>Low last 24h</strong>:{' '}
          {difference(cryptoHistory.lowLast24, cryptoHistory.low)}%
        </p>
        <p>
          <strong>Open last 24h</strong>:{' '}
          {difference(cryptoHistory.openLast24, cryptoHistory.open)}%
        </p>
      </div>
    </div>
  );
}
