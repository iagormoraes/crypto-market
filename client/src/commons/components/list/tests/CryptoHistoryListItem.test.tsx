import { render } from '@testing-library/react';

import CryptoHistoryListItem from '../CryptoHistoryListItem';

import { CryptoAsset } from '../../../../modules/crypto/interfaces/asset.interface';

describe('<CryptoHistoryListItem/>', () => {
  const mock: CryptoAsset = {
    ask: '1000.00',
    assetCode: '1000.00',
    bid: '1000.00',
    close: '1000.00',
    high: '1000.00',
    low: '1000.00',
    lowLast24: '1000.00',
    open: '1000.00',
    openLast24: '1000.00',
    volume: '1000.00',
  };

  it('matches snapshot', () => {
    const { container } = render(
      <CryptoHistoryListItem cryptoHistory={mock} />,
    );

    expect(container).toMatchSnapshot();
  });
});
