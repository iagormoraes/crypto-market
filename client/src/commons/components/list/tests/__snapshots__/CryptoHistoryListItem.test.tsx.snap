// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoHistoryListItem/> matches snapshot 1`] = `
<div>
  <div
    class="p-2 bg-blueGray-500 hover:bg-blueGray-400 transition-colors grid grid-cols-4 items-center gap-2 text-white"
  >
    <div>
      <div
        class="flex items-center gap-2"
      >
        <div
          class="w-8 h-8 cover-img rounded-full overflow-hidden shrink-0"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="bitcoin"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <h4
          class="font-bold text-lg text-white"
        >
          1000.00
        </h4>
      </div>
      <p>
        <strong>
          Low
        </strong>
        : 
        1000.00
      </p>
      <p>
        <strong>
          High
        </strong>
        : 
        1000.00
      </p>
    </div>
    <div>
      <p>
        <strong>
          Ask
        </strong>
        : $
        1000.00
      </p>
      <p>
        <strong>
          Bid
        </strong>
        : $
        1000.00
      </p>
    </div>
    <div>
      <p>
        <strong>
          Close
        </strong>
        : 
        1000.00
      </p>
      <p>
        <strong>
          Open
        </strong>
        : 
        1000.00
      </p>
    </div>
    <div>
      <p>
        <strong>
          Low last 24h
        </strong>
        :
         
        0.00
        %
      </p>
      <p>
        <strong>
          Open last 24h
        </strong>
        :
         
        0.00
        %
      </p>
    </div>
  </div>
</div>
`;
